src/gd_kotlin_configuration.cpp

```diff
@@ -0,0 +1,89 @@
+#include "gd_kotlin_configuration.h"
+#include "long_string_queue.h"
+
+#include <core/io/json.h>
+
+String GdKotlinConfiguration::to_json() {
+    Dictionary result;
+
+    String vm_type_value;
+    switch (vm_type) {
+        case jni::Jvm::Type::HOTSPOT:
+            vm_type_value = hotspot_string_identifier;
+            break;
+        case jni::Jvm::Type::GRAAL:
+            vm_type_value = graal_string_identifier;
+            break;
+        case jni::Jvm::Type::ART:
+            vm_type_value = art_string_identifier;
+            break;
+    }
+
+    result[vm_type_identifier] = vm_type_value;
+    result[max_string_size_identifier] = max_string_size;
+
+    return JSON::print(result);
+}
+
+GdKotlinConfiguration GdKotlinConfiguration::from_json(const String& json_string) {
+    Variant result;
+    String err_string;
+    int err_line;
+
+    Error error{JSON::parse(json_string, result, err_string, err_line)};
+
+    if (error != OK || result.get_type() != Variant::DICTIONARY) {
+        LOG_ERROR("Error parsing Godot Kotlin configuration ! Falling back to default configuration")
+        return GdKotlinConfiguration();
+    }
+
+    jni::Jvm::Type vm_type;
+    const Dictionary& dictionary{result.operator Dictionary()};
+#ifdef __ANDROID__
```

Comment:
Not sure if it's the responsibility of that class to control if we are in Android or not. I would just read whatever is in the file and do the android control in the gd_kotlin file.